sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/TomcatController.java

```diff
@@ -0,0 +1,188 @@
+package io.javaoperatorsdk.operator.sample;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentStatus;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.RollableScalableResource;
+import io.fabric8.kubernetes.client.dsl.ServiceResource;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import io.javaoperatorsdk.operator.api.*;
+import io.javaoperatorsdk.operator.processing.event.EventSourceManager;
+import io.javaoperatorsdk.operator.processing.event.internal.CustomResourceEvent;
+
+/**
+ * Runs a specified number of Tomcat app server Pods. It uses a Deployment to create the Pods. Also
+ * creates a Service over which the Pods can be accessed.
+ */
+@Controller
+public class TomcatController implements ResourceController<Tomcat> {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final KubernetesClient kubernetesClient;
+
+  private DeploymentEventSource deploymentEventSource;
+
+  public TomcatController(KubernetesClient client) {
+    this.kubernetesClient = client;
+  }
+
+  @Override
+  public void init(EventSourceManager eventSourceManager) {
+    this.deploymentEventSource = DeploymentEventSource.createAndRegisterWatch(kubernetesClient);
+    eventSourceManager.registerEventSource("deployment-event-source", this.deploymentEventSource);
+  }
+
+  @Override
+  public UpdateControl<Tomcat> createOrUpdateResource(Tomcat tomcat, Context<Tomcat> context) {
+    Optional<CustomResourceEvent> latestCREvent =
+        context.getEvents().getLatestOfType(CustomResourceEvent.class);
+    if (latestCREvent.isPresent()) {
```

Comment:
Using event is kinda anti-pattern, would be better to not use it in our samples, could we remove it from this controller? (2 places) ,

Reply:
Agreed, I thought about changing this but then decided to have the e2e tests first and do it a subsequent PR. But no reason we can't have that done here, especially with the tests ready.